subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -242,17 +242,24 @@ static int handle_column_status(struct bt_mesh_model *model, struct bt_mesh_msg_
 	struct bt_mesh_sensor_series_entry entry;
 
 	col_format = bt_mesh_sensor_column_format_get(type);
-
-	err = parse_series_entry(type, col_format, buf, &entry);
-	if (err == -ENOENT) {
-		/* The entry doesn't exist */
-		goto yield_ack;
+	
+	size_t val_len = sensor_value_len(type);
+	uint8_t count = buf->len / val_len;
```

Comment:
When would count be != 1?

Reply:
As far as I understand, we receive in response tuples of 3 values (if requested column exists): `{Raw_Val_X, Width, Raw_Value_Y}`. each of those values is encoded in format with certain size. Let's assume we request ID 0x0042 - Motion Sensed, as that's what PTS uses, and receive back `buf` of `len=3`, for example `000f10`. Motion Sensed has format `percentage_8` of size 1, which gives `count=3`. That is correct, and whole buffer will be decoded as follows:
- 00 -> 0.0 (X)
- 0f -> 7.5 (Width)
- 10 -> 8.0 (Y)
Without loop only X will be parsed; what's more, this will lead to not acknowledging message (as no full response was received, just like column didn't exist) and `bt_mesh_sensor_cli_series_entry_get` will return `-ETIMEDOUT`.

Reply:
No, you receive either `{PropID, Raw Value A, Raw Value B, Raw Value C}` or `{PropID, Raw Value A}`, depending on the `col_format` of the type identified by `prop_id`. This is different handling depending on `col_format` is already handled in `parse_series_entry` and is not needed here. Motion Sensed is a single channel sensor, and should return only Raw Value A (not B or C). Before this clarification of the spec leading to my changes earlier, it wasn't even clear whether or not single channel sensor types should support column/series at all, seems weird for PTS to use that ID to test this functionality?

Reply:
MshMDLd1.1r07, 4.1.5 Sensor Series Column:
> [...]
> 
> Unless otherwise specified in the Device Properties document [8], for the device properties that have one field or two fields in the Property Value and have a Sensor Update Interval value other than 0x00, the Sensor Series Column represents a single individual measurement taken over the Sensor Measurement Period interval for that device property. In such cases this state does not have Sensor Raw Value B or Raw Value C fields.